dbms/src/Interpreters/Context.cpp

```diff
@@ -1399,6 +1420,12 @@ void Context::createTMTContext(const TiFlashRaftConfig & raft_config, pingcap::C
     shared->tmt_context = std::make_shared<TMTContext>(*this, raft_config, cluster_config);
 }
 
+void Context::updateTMTContext(const TiFlashRaftConfig & raft_config, pingcap::ClusterConfig && cluster_config)
```

Comment:
I don't think we should create a new `TMTContext` just for reload config. Because most of our service objects is managed in `TMTContext` which has nothing to with TLS. And create `TMTContext` is also a heavy operation and have a high possibility to introduce other race problems at run time.

Reply:
you are right, is there a better way?

Reply:
Just reload the service related to TLS.

Reply:
> Just reload the service related to TLS.

@lidezhu 
The services related to TLS include the pd_client, rpc_client, region_cache, and lock_resolver.
I believe it is almost as heavy as reloading the TMTContext.

If it is true that other services depending on the mentioned services do not need to reload again, I can fix it by just reloading these services. For now, I am not certain about the dependencies of other services.

Reply:
For example, `KVStore` and `ManagedStorages` are services that should not be recreated when reload TLS certificate. And there are lots of other things that need to be done after create `TMTContext`, for example restore data for KVStore, register all tables in ManagedStorages and a lot more.